internal/service/chatbot/slack_workspace_data_source.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package chatbot
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/chatbot"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/chatbot/types"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkDataSource(name="Slack Workspace")
+func newDataSourceSlackWorkspace(context.Context) (datasource.DataSourceWithConfigure, error) {
+	return &dataSourceSlackWorkspace{}, nil
+}
+
+const (
+	DSNameSlackWorkspace = "Slack Workspace Data Source"
+)
+
+type dataSourceSlackWorkspace struct {
+	framework.DataSourceWithConfigure
+}
+
+func (d *dataSourceSlackWorkspace) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) { // nosemgrep:ci.meta-in-func-name
+	resp.TypeName = "aws_chatbot_slack_workspace"
+}
+
+func (d *dataSourceSlackWorkspace) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"slack_team_id": schema.StringAttribute{
+				Computed: true,
+			},
+			"slack_team_name": schema.StringAttribute{
+				Required: true,
+			},
+		},
+	}
+}
+
+func (d *dataSourceSlackWorkspace) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	conn := d.Meta().ChatbotClient(ctx)
+
+	var data dataSourceSlackWorkspaceData
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := findSlackWorkspaceByName(ctx, conn, data.SlackTeamName.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Chatbot, create.ErrActionReading, DSNameSlackWorkspace, data.SlackTeamName.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	data.SlackTeamID = flex.StringToFramework(ctx, out.SlackTeamId)
+	data.SlackTeamName = flex.StringToFramework(ctx, out.SlackTeamName)
```

Comment:
```suggestion
	resp.Diagnostics.Append(flex.Flatten(ctx, out, &data)...)
```

Since I can't readily test this, I won't make this change. However, I would encourage you to try it and, if it works, submit a PR to update this later. This simplifies code and should help make things more future proof. If it doesn't work, let us know so we can work on a fix.